src/Lean/PrettyPrinter/Delaborator/Basic.lean

```diff
@@ -216,22 +220,54 @@ where
  }
 
 def addFieldInfo (pos : Pos) (projName fieldName : Name) (stx : Syntax) (val : Expr) : DelabM Unit := do
-  let info ← mkFieldInfo projName fieldName stx val
+  let info := Info.ofFieldInfo <| ← mkFieldInfo projName fieldName stx val
   modify fun s => { s with infos := s.infos.insert pos info }
 where
-  mkFieldInfo projName fieldName stx val := return Info.ofFieldInfo {
+  mkFieldInfo projName fieldName stx val := return {
     projName := projName,
     fieldName := fieldName,
     lctx := (← getLCtx),
     val := val,
     stx := stx
   }
 
+def addOmissionInfo (pos : Pos) (stx : Syntax) (e : Expr) : DelabM Unit := do
+  let info := Info.ofOmissionInfo <| ← mkOmissionInfo stx e
+  modify fun s => { s with infos := s.infos.insert pos info }
+where
+  mkOmissionInfo stx e := return {
+    toTermInfo := ← addTermInfo.mkTermInfo stx e (isBinder := false)
+  }
+
+def withIncDepth (act : DelabM α) : DelabM α := fun ctx =>
+  act { ctx with depth := ctx.depth + 1 }
+
+def isMaxDepthReached (e : Expr) : DelabM Bool := do
```

Comment:
Should we rename this to something more general, such as `shouldOmitExpr`? In any case, let's add a docstring.
```suggestion
/--
Returns `true` if, at the current depth, we should omit the term and use `⋯` rather than delaborating it.
This function only returns `true` if `pp.omitDeepTerms` is set to `true`.
It also contains a heuristic to allow "shallow terms" to be delaborated, even if they appear deep in an expression,
which prevents terms such as atomic expressions or `OfNat.ofNat` literals from being delaborated as `⋯`.
-/
def shouldOmitExpr (e : Expr) : DelabM Bool := do
```